cmds/core/shutdown/shutdown_linux.go

```diff
@@ -41,48 +41,54 @@ var (
 		"suspend": unix.LINUX_REBOOT_CMD_SW_SUSPEND,
 		"-s":      unix.LINUX_REBOOT_CMD_SW_SUSPEND,
 	}
-	reboot = unix.Reboot
-	delay  = time.Sleep
 )
 
 func usage() {
-	log.Fatalf("shutdown [<-h|-r|-s|halt|reboot|suspend> [time [message...]]]")
+	fmt.Println("shutdown [<-h|-r|-s|halt|reboot|suspend> [time [message...]]]")
 }
 
-func main() {
-	a := os.Args
-	if len(a) == 1 {
-		a = append(a, "halt")
+func shutdown(args []string) (uint, error) {
+	if len(args) == 1 {
```

Comment:
this len(args) == 1 is then a bit less confusing, I believe:
if len(args) == 0 {
args = []string{"halt"}
}